### DISCUSSION TITLE
Instantiating `Prior` class with empirical distribution

### DISCUSSION THREAD

**Discussion by meraldoantonio** [pymc-labs/pymc-marketing#1081] [Q&A] [answered]
Is there a way to instantiate the `Prior` class using an empirical distribution, such as by passing a NumPy array?

The goal is to construct a Bayesian class that supports performing iterative Bayesian updates, where the posterior distribution from one inference can be used as the prior for the next (i.e.  continuous learning).

**Comment by williambdean** [comment]
I don't think it is supported at the moment. 

The [`Interpolated` class](https://www.pymc.io/projects/docs/en/stable/api/distributions/generated/pymc.Interpolated.html#pymc.Interpolated) would likely be route to implementing. Maybe in the way it is used [here](https://www.pymc.io/projects/examples/en/latest/howto/updating_priors.html).

Would you want to experiment with it and give an implementation a try?

Related to #782

**Comment by williambdean** [comment, marked as answer]
Hi @meraldoantonio 

We've recently made an update which allows anything that looks like the [`VariableFactory` class](https://www.pymc-marketing.io/en/latest/api/generated/pymc_marketing.prior.VariableFactory.html#pymc_marketing.prior.VariableFactory) to be used in various places. For instance, as MMM parameters

On top of the `dims` and `create_variable`, having a `to_dict` and [registered deserialization strategy](https://www.pymc-marketing.io/en/latest/api/generated/pymc_marketing.deserialize.html) should make this possible. 

Give it a try and let us know how it goes!
